apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/ContextManager.java

```diff
@@ -70,27 +70,27 @@ private static AbstractTracerContext get() {
     }
 
     /**
-     * @return the first global trace id if needEnhance. Otherwise, "N/A".
+     * @return the first global trace id when tracing. Otherwise, "N/A".
      */
     public static String getGlobalTraceId() {
         AbstractTracerContext context = CONTEXT.get();
         return Objects.nonNull(context) ? context.getReadablePrimaryTraceId() : EMPTY_TRACE_CONTEXT_ID;
     }
 
     /**
-     * @return the current segment id if needEnhance. Otherwise, "N/A".
+     * @return the current segment id when tracing. Otherwise, "N/A".
      */
     public static String getSegmentId() {
         AbstractTracerContext context = CONTEXT.get();
         return Objects.nonNull(context) ? context.getSegmentId() : EMPTY_TRACE_CONTEXT_ID;
     }
 
     /**
-     * @return the current span id if needEnhance. Otherwise, "N/A".
+     * @return the current span id when tracing. Otherwise, the value is -1.
      */
-    public static String getSpanId() {
+    public static Integer getSpanId() {
```

Comment:
The integer makes me feel, if not tracing, should null?
-1 makes sense if the return type is int.

Reply:
I struggle with that too, choose -1 to avoid null pointer at last. I would change the type to int

Reply:
It is fine to use -1 and int as return.